### PR TITLE
feat: Add a chapter on deploying and solving counter from CLI

### PR DESCRIPTION
**Pull request by mitchmindtree**

### CODE REVIEW

File: docs/getting-started/src/getting-started/counter/deploy-and-solve.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+1. **A node API** at **localhost** on **port 57967**. This allows for querying
+   the state of our local single-node blockchain.
+2. **A builder API** at **localhost** on **port 57968**. This allows for
+   submitting solutions to have them included in a block.
+
+
+## Contract Deployment
+
+Using the `essential-rest-client` tool, we can deploy our built counter
```
Comment (mohammadfawaz) [inline docs/getting-started/src/getting-started/counter/deploy-and-solve.md:44]: Should we mentioned that this should be done in separate terminal and that for that terminal, we also need the `nix` command from earlier?

File: docs/getting-started/src/getting-started/counter/deploy-and-solve.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+content address.
+
+However, this is not enough to know whether or not our solution was included in
+a block, or whether it passed the contract's constraints at all.
+
+To check whether or not our solution was successful, we can query the state of
+our contract's `counter` storage variable using the builder's node API:
+
+```console
+$ essential-rest-client query-state --content-address "1899743AA94972DDD137D039C2E670ADA63969ABF93191FA1A4506304D4033A2" "http://127.0.0.1:59438" "0000000000000000"
```
Comment (mohammadfawaz) [inline docs/getting-started/src/getting-started/counter/deploy-and-solve.md:133]:
```suggestion
$ essential-rest-client query-state --content-address "1899743AA94972DDD137D039C2E670ADA63969ABF93191FA1A4506304D4033A2" "http://127.0.0.1:57967" "0000000000000000"
```
right?

File: docs/getting-started/src/getting-started/counter/deploy-and-solve.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+```
+
+Here, we're providing the counter's contract address:
+
+```
+--content-address "1899743AA94972DDD137D039C2E670ADA63969ABF93191FA1A4506304D4033A2"
+```
+
```
Comment (mohammadfawaz) [inline docs/getting-started/src/getting-started/counter/deploy-and-solve.md:141]: We should probably also mention that we have to provide the node server API address

File: docs/getting-started/src/getting-started/counter/deploy-and-solve.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+2024-11-14T11:58:14.365887Z  INFO essential_builder_cli: Running the block builder
+```
+
+We can see that the builder exposes 2 APIs:
+
+1. **A node API** at **localhost** on **port 57967**. This allows for querying
+   the state of our local single-node blockchain.
+2. **A builder API** at **localhost** on **port 57968**. This allows for
+   submitting solutions to have them included in a block.
+
```
Comment (mohammadfawaz) [inline docs/getting-started/src/getting-started/counter/deploy-and-solve.md:40]: Maybe a note that says that we may get different ports here and to use the correct ports in the following sections and not these exact ones.

File: docs/getting-started/src/getting-started/counter/deploy-and-solve.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+## Contract Deployment
+
+Using the `essential-rest-client` tool, we can deploy our built counter
+contract to the local test builder:
+
+```console
+essential-rest-client deploy-contract "http://127.0.0.1:57968" ./out/debug/counter.json
+```
+
```
Comment (mohammadfawaz) [inline docs/getting-started/src/getting-started/counter/deploy-and-solve.md:50]: Maybe emphasize that `57968` is the port for the builder API server

**Review by mohammadfawaz — COMMENTED** [review]
Amazing! This makes things so much more approachable. Just a few nits.

File: docs/getting-started/src/getting-started/counter/deploy-and-solve.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+   submitting solutions to have them included in a block.
+
+
+## Contract Deployment
+
+Using the `essential-rest-client` tool, we can deploy our built counter
+contract to the local test builder:
+
+```console
+essential-rest-client deploy-contract "http://127.0.0.1:57968" ./out/debug/counter.json
```
[the remark this answers is not in the document]
Comment (mitchmindtree) [reply docs/getting-started/src/getting-started/counter/deploy-and-solve.md:48]: exactly

File: docs/getting-started/src/getting-started/counter/deploy-and-solve.md (hunk 2 above)
Comment (mitchmindtree) [reply docs/getting-started/src/getting-started/counter/deploy-and-solve.md:133]: wow good catch

**Comment by nfurfaro** [comment]
looks really good @mitchmindtree !
Huge simplification over the CLI approach.

File: docs/getting-started/src/getting-started/counter/deploy-and-solve.md (hunk 5 above)
Comment (mitchmindtree) [reply docs/getting-started/src/getting-started/counter/deploy-and-solve.md:58]: I think this should be OK as is, as we do mention we're deploying to the builder just above

**Review by mohammadfawaz — APPROVED** [review]
Great! :shipit:

**Comment by gweiworld** [comment]
Love!

**Pull request merged**